### PR TITLE
[GLUTEN-2661][VL] Replace unnecessary exception to fallback when use velox native parquet write

### PR DESCRIPTION
**Pull request by WolverineJiang**
## What changes were proposed in this pull request?

Some problems bring in by patch [GLUTEN-1925]. Some unnecessary exceptions are thrown and the task fails. This patch replaces the exception with a rollback.

## How was this patch tested?

Run UT by set spark.gluten.sql.native.parquet.writer.enabled=true

Run TPCDS by set spark.gluten.sql.native.parquet.writer.enabled=true && spark.sql.sources.useV1SourceList=parquet

### CODE REVIEW

**Comment by PHILO-HE** [comment]
* The title was changed to refer to issue #2661, assuming you intended to fix that issue.
* CI reported code format violation issue. Please follow the below guide to generate a fix. Thanks!
https://github.com/oap-project/gluten/blob/main/docs/developers/NewToGluten.md#javascala-code-style

**Comment by jackylee-ch** [comment]
cc @zhztheplayer @zhouyuan @binmahone

**Review by jackylee-ch — APPROVED** [review]

**Comment by binmahone** [comment]
@jackylee-ch LGTM

File: backends-velox/src/test/scala/org/apache/spark/sql/execution/VeloxParquetWriteSuite.scala [hunk 1]
```diff
@@ -70,43 +73,38 @@ class VeloxParquetWriteSuite extends WholeStageTransformerSuite {
...
-            .format("parquet")
-            .partitionBy("p")
-            .save(f.getCanonicalPath)
-        }
+        val path = f.getCanonicalPath
+        spark
+          .range(100)
+          .selectExpr("id as c1", "id % 7 as p")
+          .createOrReplaceTempView("temp")
+        val df = spark.sql(s"INSERT OVERWRITE DIRECTORY '$path' USING PARQUET SELECT * FROM temp")
```
Comment (zjuwangg) [inline backends-velox/src/test/scala/org/apache/spark/sql/execution/VeloxParquetWriteSuite.scala:93]: Here change seems loss the semantic partition by  and can not verify that the dynamiac partition write

**Pull request merged**